### ISSUE TITLE
Remove Unused Arguments Passed to Build Package Function

### ISSUE THREAD

**Issue by threeal** [threeal/CDeps.cmake#118]
This issue identifies a potential bug in the `cdeps_build_package` function, as seen in the following snippet:

https://github.com/threeal/CDeps.cmake/blame/abe0ca9e5d1e5d0a46b8de689c88e228caff17f2/cmake/CDeps.cmake#L95

The `cdeps_build_package` function is calling the `cdeps_download_package` function with additional, unparsed arguments, even though no extra arguments should be passed. This behavior likely stems from changes made before #99, where previously unparsed arguments were used to pass the `GIT_TAG` option from the `cdeps_build_package` function to the `cdeps_download_package` function.

TLDR, this issue suggests fixing the behavior by removing the unused unparsed arguments being passed to the `cdeps_download_package` function.
